api/instance.go

```diff
@@ -7,6 +7,18 @@ import (
 	"github.com/bytebase/bytebase/plugin/db"
 )
 
+// DataSourceFromInstanceWithType gets a typed data source from a instance.
+func DataSourceFromInstanceWithType(instance *Instance, dataSourceType DataSourceType) *DataSource {
```

Comment:
nit: method like this is usually put afterwards since the type definition is more important